src/Cli/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -249,18 +249,32 @@ public IReadOnlyList<FilePath> FindByPackageId(PackageId packageId)
         {
             var result = new List<FilePath>();
             bool findAnyManifest = false;
+            DirectoryPath? rootPath = null;
             foreach ((FilePath possibleManifest,
                     DirectoryPath correspondingDirectory)
                 in EnumerateDefaultAllPossibleManifests())
             {
+                if (rootPath is not null)
+                {
+                    if (!correspondingDirectory.Value.Equals(rootPath.Value))
```

Comment:
To be clear, enumerateDefaultAllPossibleManifests goes into _specific_ subdirectories; it doesn't enumerate _all_ subdirectories. It also wouldn't make sense for it to enumerate all subdirectories in this case because it's working its way up the folder hierarchy, so it would be re-searching a lot.

Reply:
Thank you for correcting that!